src/LinuxTracing/PerfEventReaders.h

```diff
@@ -55,11 +55,11 @@ SchedWakeupWithCallchainPerfEvent ConsumeSchedWakeupWithCallchainPerfEvent(
 SchedSwitchWithCallchainPerfEvent ConsumeSchedSwitchWithCallchainPerfEvent(
     PerfEventRingBuffer* ring_buffer, const perf_event_header& header);
 
-SchedSwitchWithStackPerfEvent ConsumeSchedSwitchWithStackPerfEvent(PerfEventRingBuffer* ring_buffer,
-                                                                   const perf_event_header& header);
+PerfEvent ConsumeSchedSwitchWithStackPerfEvent(PerfEventRingBuffer* ring_buffer,
+                                               const perf_event_header& header);
 
-SchedWakeupWithStackPerfEvent ConsumeSchedWakeupWithStackPerfEvent(PerfEventRingBuffer* ring_buffer,
-                                                                   const perf_event_header& header);
+PerfEvent ConsumeSchedWakeupWithStackPerfEvent(PerfEventRingBuffer* ring_buffer,
+                                               const perf_event_header& header);
```

Comment:
The names here are kind of suggesting that they produce a `SchedSwitchWithStackPerfEvent` or a `SchedWakeupWithStackPerfEvent`, as the names of these functions reflect the return values. But we know that they could also produce the "stackless" `SchedSwitchPerfEvent` and `SchedWakeupPerfEvent` (already wrapped in a `PerfEvent`).

What about calling these `ConsumeSchedSwitchWithOrWithoutStackPerfEvent` and `ConsumeSchedWakeupWithOrWithoutStackPerfEvent`? Would it help? Too ugly?

Reply:
Sounds good to me.